Sourcery/Generating/Generator.swift

```diff
@@ -101,10 +93,39 @@ extension Type {
 }
 
 enum Generator {
+    private static func updateTypeRelationship(for type: Type, typesByName: [String: Type], processed: inout [String: Bool]) {
+        type.based.keys.forEach { name in
+            guard let baseType = typesByName[name] else { return }
```

Comment:
I guess with that such test case in generator tests will fail:
`expect(generate("{% if type.Bar.based.NSObject %} TRUE {% endif %}")).to(equal(" TRUE "))`
as `NSObject` is not known type so is not in `typesByName`